src/lazy/expanded/mod.rs

```diff
@@ -289,6 +291,58 @@ impl<'top> EncodingContextRef<'top> {
             .clone_macro_with_name("values")
             .expect("`values` macro in system macro table")
     }
+
+
+    #[cfg(feature = "lazy-source-location")]
+    pub fn set_value_start(&mut self, start: usize) {
+        self.value_start = start;
+    }
+
+    #[cfg(feature = "lazy-source-location")]
+    pub fn set_value_end(&mut self, end: usize) {
+        self.value_end = end;
+    }
+
+    #[cfg(feature = "lazy-source-location")]
+    pub fn location(&self) -> Option<(usize, usize)> {
+        match unsafe { &*self.io_buffer_source.get() } {
+            IoBufferSource::IoBuffer(ref buffer) => Some((buffer.row(), buffer.column())),
+            IoBufferSource::Reader(handle) => {
+                let prev_newline = handle.prev_newline_offset();
+                let current_row = handle.row();
+
+                if prev_newline == 0 {
+                    // If no newlines have been encountered yet, we're on the first row
+                    // The column is the start position of the value
+                    Some((1, self.value_start))
+                } else {
+                    // If newlines have been encountered
+                    if self.value_start > prev_newline {
+                        // If the value starts after the last newline,
+                        // we're on the current row and can calculate the column
+                        Some((current_row, self.value_start - prev_newline))
+                    } else {
```

Comment:
I believe all of this is actually unnecessary for correctness. The method you have for calculating the row/column on lines 325-337 should work for these cases too. Is there a noticeable performance benefit from having these special cases up front? (If not, then can you simplify it by removing the special cases?)

Reply:
Actually this is necessary for correctness. We have two implementations of `IobufferHanlde` one with `IonSlice` and one with `IonStream`. the `IonStream` one reuses its buffer and hence we have to keep track of the value_start and value_end in order to decide the correct location info for it.
If we remove these lines that you mentioned are unnecessary the tests we have in `LazyValue` will fail, due to incorrect count of location.

e.g.
```
    #[case::lf_lf_cr("{foo: 1, bar: 2}\n\n\r\"hello\"", (4,1))]
```
 fails with
 ```
   left: (1, 20)
   right: (4, 1)
```